Python/_warnings.c

```diff
@@ -254,6 +262,69 @@ warnings_lock_held(WarningsState *st)
     return PyMutex_IsLocked(&st->lock.mutex);
 }
 
+static PyObject *
+get_warnings_context(PyInterpreterState *interp)
+{
+    WarningsState *st = warnings_get_state(interp);
+    assert(PyContextVar_CheckExact(st->context));
+    PyObject *ctx;
+    if (PyContextVar_Get(st->context, NULL, &ctx) < 0) {
+        return NULL;
+    }
+    if (ctx == NULL) {
+        Py_RETURN_NONE;
+    }
+    return ctx;
+}
+
+static PyObject *
+get_warnings_context_filters(PyInterpreterState *interp)
+{
+    PyObject *ctx = get_warnings_context(interp);
+    if (ctx == NULL) {
+        return NULL;
+    }
+    if (ctx == Py_None) {
+        Py_DECREF(ctx);
+        Py_RETURN_NONE;
+    }
+    PyObject *context_filters = PyObject_GetAttrString(ctx, "_filters");
```

Comment:
Using static strings would be more performant
```suggestion
    PyObject *context_filters = PyObject_GetAttr(ctx, &_Py_ID(_filters));
```